client/components/material-icon/index.tsx

```diff
@@ -3,10 +3,19 @@
  */
 import classnames from 'classnames';
 import React from 'react';
-import PropTypes from 'prop-types';
 import spritePath from '@automattic/material-design-icons/svg-sprite/material-icons.svg';
 
-function MaterialIcon( props ) {
+type Omit< T, K extends keyof any > = Pick< T, Exclude< keyof T, K > >;
```

Comment:
Was this unavailable? We do have `utility-types` which can provide this:

```ts
import { Omit } from 'utility-types'
```

Reply:
`utility-types` lead me to [`Overwrite`](https://github.com/piotrwitek/utility-types#overwritet-u) - that might be a good solution to the ugly extend.

Reply:
Oh! I didn't know about 'utility-types', thanks!

Since I use VS Code, it's hard for me to tell if something only works because of it being available in the IDE's TS.

Reply:
Right, looks like `Assign` does the job. Thanks again! 😁